### PR TITLE
SNOW-1921713 Encode large integers in AST

### PR DESCRIPTION
**Pull request by sfc-gh-vbudati**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-1921713

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://github.com/snowflakedb/snowpark-python/blob/main/CONTRIBUTING.md#thread-safe-development)

3. Please describe how your code solves the related issue.

Encode large integers (larger than 64 bits) in AST. Related design doc: https://docs.google.com/document/d/1brLIUPZumyoqyY5posr4m_lJDV6d2jU37ZnqtKigU9Y/edit?usp=sharing 
I also added `_emit_ast=False` for some APIs called with `to_snowpark_pandas` (this is in session.py).

### CODE REVIEW

File: src/snowflake/snowpark/_internal/ast/utils.py [hunk 1]
```diff
@@ -221,7 +221,14 @@ def build_expr_from_python_val(
 
     elif isinstance(obj, int):
         ast = with_src_position(expr_builder.int64_val)  # type: ignore[arg-type] # TODO(SNOW-1491199) # Argument 1 to "with_src_position" has incompatible type "Int64Val"; expected "Expr"
-        ast.v = obj  # type: ignore[attr-defined] # TODO(SNOW-1491199) # "Expr" has no attribute "v"
+        # If the integer is too large to fit in 64 bits, we need to convert it to bytes.
+        # The absolute value of the integer needs to be encoded in big-endian mode.
+        if obj.bit_length() > 64:
```
Comment (sfc-gh-azwiegincew) [inline src/snowflake/snowpark/_internal/ast/utils.py:226]: I think this should be `>=` to account for signed ints.

File: tests/ast/data/Session.create_dataframe.test [hunk 2]
```diff
@@ -24,6 +24,10 @@ df8 = session.create_dataframe([Row(a=1, b=2, c=3, d=4)])
 
 df9 = session.create_dataframe([{"a": 1}, {"b": 2}])
 
+df10 = session.create_dataframe([2**100, 2**100])
```
Comment (sfc-gh-azwiegincew) [inline tests/ast/data/Session.create_dataframe.test:27]: Maybe try `2**64 – 1` as well.

**Review by sfc-gh-azwiegincew — APPROVED** [review]

File: src/snowflake/snowpark/_internal/ast/utils.py [hunk 3]
```diff
@@ -236,7 +236,14 @@ def build_expr_from_python_val(
 
     elif isinstance(obj, int):
         ast = with_src_position(expr_builder.int64_val)  # type: ignore[arg-type] # TODO(SNOW-1491199) # Argument 1 to "with_src_position" has incompatible type "Int64Val"; expected "Expr"
```
Comment (sfc-gh-heshah) [inline src/snowflake/snowpark/_internal/ast/utils.py:238]: Nit: move to the else block below.

File: tests/ast/data/Session.create_dataframe.test [hunk 4]
```diff
@@ -24,6 +24,10 @@ df8 = session.create_dataframe([Row(a=1, b=2, c=3, d=4)])
 
 df9 = session.create_dataframe([{"a": 1}, {"b": 2}])
 
+df10 = session.create_dataframe([2**100, 2**100, 2**64 - 1, 2**64, 1, 2**64 - 2])
+
+df11 = session.create_dataframe([[2**100, 2**100], [2**100, 2**100]])
+
```
Comment (sfc-gh-heshah) [inline tests/ast/data/Session.create_dataframe.test:30]: Can you please add expectation tests for negative values as well?

**Review by sfc-gh-heshah — CHANGES_REQUESTED** [review]

File: tests/ast/data/Session.create_dataframe.test (hunk 4 above)
Comment (sfc-gh-vbudati) [reply tests/ast/data/Session.create_dataframe.test]: done!

File: src/snowflake/snowpark/_internal/ast/utils.py (hunk 3 above)
Comment (sfc-gh-vbudati) [reply src/snowflake/snowpark/_internal/ast/utils.py]: done!
